packages/http-server-csharp/src/lib/interfaces.ts

```diff
@@ -77,7 +77,18 @@ export class CSharpType implements CSharpTypeMetadata {
     }
   }
   public getTypeReference(scope?: Scope<string>): string {
-    return `${this.isNamespaceInScope(scope) ? "" : this.namespace + "."}${this.name}`;
+    if (this.isNamespaceInScope(scope)) {
+      return this.name;
+    }
+
+    const collectionPatternRegex = /^(ISet|IEnumerable|ICollection)<(.+)>$/;
+    const collectionPattern = this.name.match(collectionPatternRegex);
+    if (collectionPattern) {
+      const [_, collectionType, innerType] = collectionPattern;
+      return `${collectionType}<${this.namespace + "." + innerType}>`;
```

Comment:
I am a little uncomfortable with this, I feel like, since we are creating these types, we can add properties at the time of creation..  Whether this means having a CSharpCollectionType type (or similar) that extends CSharpType and uses its own properties, or something else.  It feels wrong to use a regex to interpret a type that we created.